### PR TITLE
Update Pedigree class (pedigree_v2) and test files. 

### PR DESCRIPTION
**Pull request by stevenhwu**
This will be a drop-in replace of the existing class.

The update version will allow more detailed testings, and might be easier to handle different peeling operations if required.
Currently, this class (pedigree_v2) is not used in call.cc yet.

test/CMakeLists.txt might cause conflict later, because not all library are included.
test/src/fixture/ include fixture for boot_test that shared between multiple tests.

### CODE REVIEW

**Comment by reedacartwright** [comment]
Jenkins, ok to test.

File: src/include/utils/boost_utils.h [hunk 1]
```diff
@@ -0,0 +1,23 @@
+//
+// Created by steven on 2/4/16.
+//
+
```
Comment (reedacartwright) [inline src/include/utils/boost_utils.h:4]: Missing boost headers

File: src/include/utils/assert_utils.h [hunk 2]
```diff
@@ -0,0 +1,70 @@
+//
+// Created by steven on 1/29/16.
+//
+
+#ifndef DENOVOGEAR_ASSERT_UTILS_H
+#define DENOVOGEAR_ASSERT_UTILS_H
+
+#include <cassert>
+
```
Comment (reedacartwright) [inline src/include/utils/assert_utils.h:9]: Please wrap these functions such that if NDEBUG is defined they evaluate to no-ops.

File: src/include/utils/assert_utils.h [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+const double ASSERT_CLOSE_THRESHOLD = 1e-6;
+
+//TODO: HACK: FIXME:
+template<typename A, typename B>
+void AssertEqual(A expected, B actual) {
+    assert(expected == actual);
+};
+
```
Comment (reedacartwright) [inline src/include/utils/assert_utils.h:19]: Also, all these function names need to be in lower_case: e.g. `assert_equal`.

**Comment by reedacartwright** [comment]
It is hard to see what is being changed since you put the new code in different files.

File: src/include/utils/boost_utils.h (hunk 1 above)
Comment (stevenhwu) [inline src/include/utils/boost_utils.h:4]: Updated at the next commit.
Both this and call.cc can compile and run without any #include <boost/**>. Is this a bad sign?

**Comment by stevenhwu** [comment]
Update some of these comments.
Let's discuss about "It is hard to see what is being changed since you put the new code in different files." later.

**Comment by stevenhwu** [comment]
Can jenkin ignore this PR please.

**Pull request closed without merging**